telebot/async_telebot.py

```diff
@@ -183,6 +189,13 @@ def __init__(self, token: str, parse_mode: Optional[str]=None, offset: Optional[
         self.middlewares = []
 
         self._user = None # set during polling
+        self.bot_id: int = None
+
+        if validate_token:
+            util.validate_token(self.token)
+        
+        self.bot_id: int = util.extract_bot_id(self.token) # subject to change, unspecified
```

Comment:
May cause error if token is not in valid form. extract_bot_id should return NONE if extract fails.